transform/app/api/routes/solr_utils/solr.py

```diff
@@ -0,0 +1,65 @@
+from fastapi import APIRouter, Query
+from datetime import datetime
+import requests
+
+from app.transform.utils.loader import load_env_vars
+from app.transform.schemas.properties.env import (
+    SOLR_ADDRESS,
+    SOLR_PORT,
+    SOLR_COLLECTION_NAMES,
+)
+
+env_vars = load_env_vars()
+router = APIRouter()
+
+
+@router.post("/create_collections")
+async def create_solr_collections(
+    all_collection_config: str = Query(
+        ...,
+        description="Config name for 'all_collection'",
+        example="all_collection_oag56_v203",
+    ),
+    provider_config: str = Query(
+        ..., description="Config name for 'provider'", example="provider_v101"
+    ),
+    collection_prefix: str = Query(
+        "",
+        description="Prefix for collection names. "
+        "It is recommended to use convention 'oag<dump_version>'",
+        example="oag56",
+    ),
+    date: str = Query(
+        None,
+        description="Date string in the format 'YYYYMMDD'. "
+        "Defaults to the current date if not provided.",
+        example="20231114",
+    ),
+    num_shards: int = Query(1, description="Number of shards", example=1),
+    replication_factor: int = Query(1, description="Replication factor", example=1),
+):
+    """
+    Creates Solr collections for a singular data iteration.
+    """
+    date = date or datetime.now().strftime("%Y%m%d")
+
+    for collection in SOLR_COLLECTION_NAMES:
```

Comment:
I think it should be a celery task because it takes a while (like 10 seconds or more) to be completed. 
It should be done async and task id should be returned. 
At least you should be able to see the task status in a flower dashboard

Reply:
I added it to the Celery task and I moved the batch task to the `tasks/batch.py` as indicated by the comment `TODO move batch task here`